catalog/plugin/catalog-plugin-query-monitor-plugin/src/main/java/ddf/catalog/querymonitor/QueryMonitorPluginImpl.java

```diff
@@ -0,0 +1,170 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.querymonitor;
+
+import java.util.HashMap;
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.operation.QueryRequest;
+
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostFederatedQueryPlugin;
+import ddf.catalog.plugin.PreFederatedQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.source.Source;
+
+public class QueryMonitorPluginImpl
+
+        implements QueryMonitorPlugin, PreFederatedQueryPlugin, PostFederatedQueryPlugin {
+
+    boolean removeFromTable = true;
+
+    private HashMap<UUID, ActiveSearch> activeSearchTable = new HashMap<UUID, ActiveSearch>();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueryMonitorPluginImpl.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public HashMap<UUID, ActiveSearch> getSearchTable() {
+        return activeSearchTable;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setRemoveFromTable(boolean b) {
+        removeFromTable = b;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean addActiveSearch(ActiveSearch as) {
+        if (as == null) {
+            LOGGER.warn("Cannot add null ActiveSearch to table!");
+            return false;
+        }
+        activeSearchTable.put(as.getUniqueID(), as);
+        return true;
+    }
+
+    public QueryMonitorPluginImpl() {
+        LOGGER.info("QueryMonitor started");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean removeActiveSearch(UUID uniqueID) {
+        if (uniqueID == null) {
+            LOGGER.warn("Can't remove active search with null ID.");
+            return false;
+        }
+        if (activeSearchTable == null) {
+            LOGGER.warn("Can't remove active search. Search not found in table.");
+            return false;
+        }
+        if (activeSearchTable.remove(uniqueID) != null) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Method that is implemented for {@link PreFederatedQueryPlugin}. Uses the given {@link Source}
+     * and {@link QueryRequest} information to create a new {@link ActiveSearch} to add to the ActiveSearchTable.
+     *
+     * @param source {@link Source} that corresponds to source the search is querying
+     * @param input  {@link QueryRequest} that corresponds to request generated when a user queried the source
+     * @return {@link QueryRequest} that was given as a parameter with updated property information corresponding
+     * to the {@link ActiveSearch}'s {@link UUID}
+     */
+
+    @Override
+    public QueryRequest process(Source source, QueryRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        try {
+            if (source == null) {
```

Comment:
This looks like object construction code.  I think you should push most of this implementation down into an ActiveSearch constructor that takes a `QueryRequest` parameter.